docs/assembly.rst

```diff
@@ -96,6 +96,26 @@ you really know what you are doing.
                 }
             }
         }
+
+        // Same as above, but accomplish the entire code within inline assembly.
+        function sumPureAsm(uint[] _data) returns (uint o_sum) {
+            assembly {
+               // Load the length (first 32 bytes)
+               let len := mload(_data)
+
+               // Skip over the length field.
+               _data := add(_data, 0x20)
+
+               // Iterate until the bound is not met.
+               for
+                   { let end := add(_data, len) }
+                   lt(_data, end)
+                   { _data := add(_data, 0x20) }
```

Comment:
Perhaps a comment would be enough to clarify that - the main goal of that example is how to really on assembly only.